### PR TITLE
wiringpi: add version 3.2

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **wiringpi/3.2**

https://github.com/WiringPi/WiringPi/compare/2.61-1...3.2

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — DISMISSED** [review]
Nice to see this project active again!

File: recipes/wiringpi/all/conanfile.py [hunk 1]
```diff
@@ -32,16 +38,24 @@ def configure(self):
...
         if self.settings.os != "Linux":
             raise ConanInvalidConfiguration(f"{self.ref} only works for Linux")
-
-    def layout(self):
-        cmake_layout(self, src_folder="src")
+        if Version(self.version) >= 3.0:
+            if self.settings.compiler == "gcc" and \
+                Version(self.settings.compiler.version) < 8:
+                raise ConanInvalidConfiguration(f"{self.ref} requires gcc >= 8")
+            if self.settings.compiler == "gcc" and \
```
Comment (uilianries) [inline recipes/wiringpi/all/conanfile.py:51]:
```suggestion
            # wiringPi.c:1755:9: error: case label does not reduce to an integer constant
            if self.settings.compiler == "gcc" and \
```

File: recipes/wiringpi/all/conanfile.py [hunk 2]
```diff
@@ -32,16 +38,24 @@ def configure(self):
...
+
     def validate(self):
         if self.settings.os != "Linux":
             raise ConanInvalidConfiguration(f"{self.ref} only works for Linux")
-
-    def layout(self):
-        cmake_layout(self, src_folder="src")
+        if Version(self.version) >= 3.0:
+            if self.settings.compiler == "gcc" and \
+                Version(self.settings.compiler.version) < 8:
```
Comment (uilianries) [inline recipes/wiringpi/all/conanfile.py:49]: Why does it need GCC 8 ?

Comment (toge) [reply recipes/wiringpi/all/conanfile.py:53]: @uilianries 
using const variables are not valid in these versions.

```
src/wiringPi/wiringPi.c:144:39: error: initializer element is not constant
 const unsigned int RP1_INV_PUD_MASK = ~(RP1_PUD_UP | RP1_PUD_DOWN); //~0x0C
                                       ^
```

Comment (uilianries) [reply recipes/wiringpi/all/conanfile.py:53]: @toge Thank you for pointing! 😄

**Review by uilianries — APPROVED** [review]
LGTM

**Review by jwillikers — APPROVED** [review]

**Pull request merged**